File: clojure/resistor-color-duo/src/resistor_color_duo.clj
(ns resistor-color-duo)

(def color->code {"black" 0, "brown" 1, "red" 2, "orange" 3, "yellow" 4,
                  "green" 5, "blue" 6, "violet" 7, "grey" 8, "white" 9})

(defn resistor-value
  "Returns the resistor value based on the given colors."
  [colors]
  (parse-long (apply str (take 2 (map color->code colors)))))

File: elixir/resistor-color-duo/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @bands %{
    black: 0, brown: 1, red: 2, orange: 3, yellow: 4,
    green: 5, blue: 6, violet: 7, grey: 8, white: 9
  }

  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    [tens, units | _] = Enum.map(colors, &(@bands[&1]))
    10 * tens + units
  end
end
